dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMain.java

```diff
@@ -88,6 +88,23 @@ public static void main(String[] arguments)
             inputClassPath, entryPoints, linkageCheckerArguments.getInputExclusionFile());
     ImmutableSet<LinkageProblem> linkageProblems = linkageChecker.findLinkageProblems();
 
+    linkageProblems = filterReachable(linkageCheckerArguments, linkageChecker, linkageProblems);
+
+    if (writeExclusionFile(linkageCheckerArguments, linkageProblems)) {
+      return ImmutableSet.of();
+    }
+
+    if (!linkageProblems.isEmpty()) {
+      System.out.println(LinkageProblem.formatLinkageProblems(linkageProblems));
+    }
+
+    return linkageProblems;
+  }
+
+  private static ImmutableSet<LinkageProblem> filterReachable(
+      LinkageCheckerArguments linkageCheckerArguments,
+      LinkageChecker linkageChecker,
+      ImmutableSet<LinkageProblem> linkageProblems) {
     if (linkageCheckerArguments.getReportOnlyReachable()) {
```

Comment:
I'm tempted to pull the if out of this method and not pass the linkageChecker arguments in